src/partition/repart.c

```diff
@@ -561,6 +566,12 @@ static Partition* partition_free(Partition *p) {
 
         partition_encrypted_volume_free(p->encrypted_volume);
 
+        if (p->fallback_target_for)
+                p->fallback_target_for->fallback_for = NULL;
```

Comment:
i'd always add an assert() that validates the backpointer to us first, just so that we catch incorrect setups instead of just walzing all over them.